osu.Game/Skinning/SkinnableSound.cs

```diff
@@ -79,7 +75,7 @@ public SkinnableSound([NotNull] ISampleInfo sample)
         /// <summary>
         /// The samples that should be played.
         /// </summary>
-        public ISampleInfo[] Samples
+        public ISampleInfo[]? Samples
```

Comment:
This is a bit awful... The property can accept `null` to unset the samples (and this is used in places), but will never *return* null. Which is tripping a code quality inspection as well (albeit in a test scene).

Don't necessarily have any good ideas what to do here. Simplest thing would be to just make the test shut up. But it'd be nice to actually have this *not* nullable - maybe there could be a `ClearSamples()` convenience method or something?

Reply:
Yeah I was worried about this, but didn't want to expand the scope of this PR further. I'll see if the non-nullable route requires minimal changes.